src/Board.java

```diff
@@ -97,4 +106,40 @@ private void updatePieceCount(int position) {
             this.numberOfBlackPieces--;
         }
     }
+    
+    public boolean isEndState() {
+        return (numberOfBlackPieces == 0 || numberOfWhitePieces == 0);
+    }
+
+    public List<Square> getAdjacentSquares(Square square) {
+        List<Integer> squareNumbers = square.getAdjacentSquares();
+        return getSquares((ArrayList<Integer>) squareNumbers);
+    }
+
+    public List<Square> getSquaresTwoMovesAway(Square square) {
```

Comment:
So I believe this method is intended to find the single jump locations, however a simpler way of doing this is just saying ± 7 and ± 9 from the starting position. The MoveValidator handles the checks for illegal backwards moves and jumping off the edge of the board.